src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java

```diff
@@ -151,7 +151,7 @@ public class GitHubSCMSource extends AbstractGitSCMSource {
 
   public static final String VALID_GITHUB_REPO_NAME = "^[0-9A-Za-z._-]+$";
   public static final String VALID_GITHUB_USER_NAME =
-      "^[A-Za-z0-9](?:[A-Za-z0-9]|-(?=[A-Za-z0-9])){0,38}$";
+      "^[A-Za-z0-9](?:[A-Za-z0-9_]|-(?=[A-Za-z0-9])){0,38}$";
```

Comment:
To my understanding, this regexp is not 100% accurate as GitHub username in the context of EMU can contain only one underscore. It is not stated exactly like this, but the username are suffixed with `_SHORT-CODE` and the short code is a `a three-to-eight character alphanumeric string, and contain no special characters`.
So that fixes the problem but also does passes values that are now actually allowed.

Reply:
Your suggestion is excellent, I've tested it and that works with all the additional testcase expressions added, thanks!